metaflow/plugins/pypi/pip.py

```diff
@@ -258,6 +258,15 @@ def indices(self, prefix):
         return index, extras
 
     def _call(self, prefix, args, env=None, isolated=True):
+        # `--no-input` flag should not be passed when credentials
```

Comment:
can you go into a bit more detail why removing the `--no-input` was necessary? we don't want to require/enable any user interaction with the pip3 calls.

Reply:
Honestly, I am not sure. What I observed was that perhaps keyring tries to inject the creds dynamically using the json file passed to the env variable. And that apparently doesn't work when we use `--no-input`

Reply:
this will likely break the pypi solver. can we roll back this PR and discuss this issue?